Sources/Subscription/DeadTokenRecoverer.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  DeadTokenRecoverer.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Networking
+import os.log
+
+@available(macOS 12.0, *)
+public struct DeadTokenRecoverer {
+
+    private static var recoveryAttempted: Int = 0
+
+    public static func recoverDeadToken(endpointService: any SubscriptionEndpointServiceV2,
```

Comment:
Two things:
1. I am thinking about more descriptive name for it as it does very specific thing. I am missing that this is an attempt (and may fail) and how the recovery is made. Suggestions: 
- `attemptDeadTokenRecoveryFromPastPurchase(endpointService: any SubscriptionEndpointServiceV2, restoreFlow: any AppStoreRestoreFlowV2)`
- `attemptDeadTokenRecovery(using restoreFlow: any AppStoreRestoreFlowV2, endpointService: any SubscriptionEndpointServiceV2)`
2. To be wary for the "real" implementation if the function can potentially be invoked multiple times in parallel without synchronisation.

Reply:
1. What about `DeadTokenRecoverer.attemptRecoveryFromPastPurchase(...)`?
2. In theory, `recoveryAttemptCount` in the struct is the safeguard for that. Only one recovery operation at the time will be executed, the other ones will throw a `.tokenUnRefreshable`